frappe/automation/doctype/auto_repeat/auto_repeat.py

```diff
@@ -282,32 +292,33 @@ def notify_error_to_user(self, error_log):
 		)
 
 
-def get_next_schedule_date(schedule_date, frequency, start_date, repeat_on_day=None, repeat_on_last_day=False, end_date=None, for_full_schedule=False):
-	if month_map.get(frequency):
-		month_count = month_map.get(frequency) + month_diff(schedule_date, start_date) - 1
+def get_next_schedule_date(schedule_date, auto_repeat_doc=None, for_full_schedule=False):
```

Comment:
It seems that this function is used in this file alone and requires an Auto Repeat document to retrieve information too. Why not add it as a method in the  `Auto Repeat` class?